File: src/main/java/com/cpw/browser/action/GoHomeAction.java
package com.cpw.browser.action;

import com.cpw.browser.WebBrowserIcons;
import com.cpw.browser.settings.BrowserSettingsState;
import com.cpw.browser.toolwindow.BrowserTabManager;
import com.cpw.browser.toolwindow.BrowserTabPanel;
import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;

// 导航到主页的 Action
public class GoHomeAction extends AnAction {

    private final BrowserTabManager tabManager;

    public GoHomeAction(BrowserTabManager tabManager) {
        super("主页", "回到主页", WebBrowserIcons.HOME);
        this.tabManager = tabManager;
    }

    @Override
    public void update(AnActionEvent e) {
        // 当存在活动标签页时启用
        e.getPresentation().setEnabled(tabManager.getActiveTab() != null);
    }

    @Override
    public void actionPerformed(AnActionEvent e) {
        BrowserSettingsState settings = BrowserSettingsState.getInstance();
        // 如果主页 URL 为空则使用 about:blank
        String homeUrl = settings.getHomePageUrl().isBlank() ? "about:blank" : settings.getHomePageUrl();
        BrowserTabPanel activeTab = tabManager.getActiveTab();
        if (activeTab != null) {
            activeTab.navigate(homeUrl);
        }
    }
}

File: src/main/kotlin/com/cpw/browser/action/GoHomeAction.kt
package com.cpw.browser.action

import com.cpw.browser.WebBrowserIcons
import com.cpw.browser.browser.BrowserTabManager
import com.cpw.browser.settings.BrowserSettingsState
import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent

class GoHomeAction(private val tabManager: BrowserTabManager) : AnAction("主页", "回到主页", WebBrowserIcons.Home) {

    override fun update(e: AnActionEvent) {
        e.presentation.isEnabled = tabManager.activeTab != null
    }

    override fun actionPerformed(e: AnActionEvent) {
        val settings = BrowserSettingsState.getInstance()
        val homeUrl = settings.homePageUrl.ifBlank { "about:blank" }
        tabManager.activeTab?.navigate(homeUrl)
    }
}
